ci: use Standard Library in Jenkinsfile to link sub-projects together

thst-nordic opened the pull request:
Managing Docker version, Agent Labels, Github Status, manifest updates, downstream jobs from CI_LIB.

More info: https://projecttools.nordicsemi.no/confluence/display/NCS/Linking+Jenkins+Jobs+Together

Signed-off-by: Thomas Stilwell Thomas.Stilwell@nordicsemi.no

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/NordicPlayground/fw-nrfconnect-nrf?pullRequest=979) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/NordicPlayground/fw-nrfconnect-nrf?pullRequest=979) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/NordicPlayground/fw-nrfconnect-nrf?pullRequest=979) it.</sub>

lemrey commented:
Why is there a `[nrf noup]` tag in the commit?

NordicBuilder commented:
Found the following issues, please fix and resubmit:

## checkpatch issues

```
No additional types will be considered - file 'scripts/checkpatch/typedefsfile': No such file or directory
-:90: ERROR:TRAILING_WHITESPACE: trailing whitespace
#90: FILE: Jenkinsfile:59:
+          if (lib_State.isOrphanJob(CI_STATE)){ // Does the JOB have a parent? $

-:111: ERROR:TRAILING_WHITESPACE: trailing whitespace
#111: FILE: Jenkinsfile:70:
+        lib_West.ApplyManifestUpdates(CI_STATE) $

-:159: ERROR:TRAILING_WHITESPACE: trailing whitespace
#159: FILE: Jenkinsfile:163:
+              build job: "${it}", propagate: CI_STATE.NRF.WAITING, wait: CI_STATE.NRF.WAITING, $

-:199: ERROR:TRAILING_WHITESPACE: trailing whitespace
#199: FILE: Jenkinsfile:200:
+  * $

-:202: ERROR:TRAILING_WHITESPACE: trailing whitespace
#202: FILE: Jenkinsfile:203:
+  * $

- total: 5 errors, 0 warnings, 206 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

NOTE: Whitespace errors detected.
      You may wish to use scripts/cleanpatch or scripts/cleanfile

Your patch has style problems, please review.

NOTE: Ignored message types: AVOID_EXTERNS BRACES CONFIG_EXPERIMENTAL CONST_STRUCT DATE_TIME FILE_PATH_CHANGES MINMAX MULTISTATEMENT_MACRO_USE_DO_WHILE NETWORKING_BLOCK_COMMENT_STYLE PRINTK_WITHOUT_KERN_LEVEL SPLIT_STRING VOLATILE

NOTE: If any of the errors are false positives, please report
      them to the maintainers.



```
## Gitlint issues

3: UC4 Line exceeds max length (100>72): "Managing Docker version, Agent Labels, Github Status, manifest updates, downstream jobs from CI_LIB."

thst-nordic commented:
[nrf noup] has been removed.  

These PR's all have very similiar changes in order to link the CI plans together:     All are passing CI. 
https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/979
https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/174
https://github.com/NordicPlayground/fw-nrfconnect-mcuboot/pull/45
https://github.com/NordicPlayground/nrfxlib/pull/76

thst-nordic commented:
I updated the west.yml, and CI is running.

Regarding commit statuses:  these are deprecated
 - continuous-integration/jenkins/doc
 - continuous-integration/jenkins/pr-head

Replaced with:
 - CI/jenkins/NRF
 - CI/jenkins/DOC
 - CI/jenkins/TEST_MCUBOOT
 - CI/jenkins/DESKTOP
